src/Image.cc

```diff
@@ -29,15 +48,55 @@ struct ArrayDeleter
   }
 };
 
+//////////////////////////////////////////////////
+Image::Image()
+  : dataPtr(utils::MakeUniqueImpl<Implementation>())
+{
+}
+
 //////////////////////////////////////////////////
 Image::Image(unsigned int _width, unsigned int _height,
-  PixelFormat _format) :
-  width(_width),
-  height(_height)
+  PixelFormat _format)
+  : dataPtr(utils::MakeUniqueImpl<Implementation>())
 {
-  this->format = PixelUtil::Sanitize(_format);
+  this->dataPtr->width = _width;
+  this->dataPtr->height = _height;
+  this->dataPtr->format = PixelUtil::Sanitize(_format);
   unsigned int size = this->MemorySize();
-  this->data = DataPtr(new unsigned char[size], ArrayDeleter<unsigned char>());
+  this->dataPtr->data =
+      DataPtr(new unsigned char[size], ArrayDeleter<unsigned char>());
```

Comment:
Prefer std::make_shared

Reply:
I think `std::make_shared` is not able to take a custom deleter as arg?

Reply:
Ah you are correct.